floracast/steps/train.py

```diff
@@ -0,0 +1,133 @@
+"""
+Model training step for FloraCast.
+"""
+
+from typing import Annotated
+from darts import TimeSeries
+from darts.models import TFTModel, ExponentialSmoothing
+from zenml import step
+from zenml.logger import get_logger
+from materializers.tft_materializer import (
+    TFTModelMaterializer,
+)  # Import for explicit usage
+
+logger = get_logger(__name__)
+
+
+@step(output_materializers={"trained_model": TFTModelMaterializer})
+def train_model(
+    train_series: TimeSeries,
+    model_name: str = "TFTModel",
+    input_chunk_length: int = 30,
+    output_chunk_length: int = 7,
+    hidden_size: int = 32,
+    lstm_layers: int = 1,
+    num_attention_heads: int = 2,
+    dropout: float = 0.1,
+    batch_size: int = 16,
+    n_epochs: int = 20,
+    random_state: int = 42,
+    add_relative_index: bool = True,
+    enable_progress_bar: bool = False,
+    enable_model_summary: bool = False,
+) -> Annotated[TFTModel, "trained_model"]:
```

Comment:
```suggestion
) -> Annotated[Union[TFTModel, ExponentialSmoothing], "trained_model"]:
```

probably rather this (and then you need to import `Union` from `typing` above too.